src/RankedPairsCalculator.php

```diff
@@ -36,71 +41,55 @@ public function __construct(Ballot $tieBreakingBallot)
             $ballotTieBreaker = new BallotTieBreaker();
             $myTieBreakingBallot = $ballotTieBreaker->breakTiesRandomly($myTieBreakingBallot);
         }
+        $this->tieBreakingBallot = $myTieBreakingBallot;
+
         $tieBreaker = new TotallyOrderedBallotPairTieBreaker(new CandidateComparator($myTieBreakingBallot));
         $this->tieBreakingPairComparator = new TieBreakingPairComparator($tieBreaker);
         $this->tieBreakingCandidateComparator = new CandidateComparator($myTieBreakingBallot);
+
+        $this->marginRegistrar = new MarginRegistrar();
+        $this->winningVotesRegistrar = new WinningVoteRegistrar();
     }
 
     /**
      * @param int number of winners to return.
-     * @param NBallot the ballots from the electorate.
-     * @return CandidateList in which the zeroth Candidate is the most preferred, the first Candidate is the second most
-     * preferred, and so on until the last Candidate who is the least preferred. Ties are broken according to the Ballot
-     * provided to the constructor.
+     * @param NBallot[] $nBallots
+     * @return Result
      */
-    public function calculate(int $numWinners, NBallot ...$nBallots) : CandidateList
+    public function calculate(int $numWinners, NBallot ...$nBallots) : Result
     {
         $agenda = new Agenda(...$nBallots);
```

Comment:
Would it make sense to pass in an agenda rather than a list of ballots?  Do we want the list of candidates determined on the pivot side, or tideman side?

Reply:
Hmmm... good idea. How about we default to the library side, but give the app the ability to override?